src/Kestrel.Core/Internal/Http2/Http2PeerSettings.cs

```diff
@@ -91,16 +85,42 @@ public void ParseFrame(Http2Frame frame)
             }
         }
 
-        public IEnumerator<Http2PeerSetting> GetEnumerator()
+        // Gets the settings that are different from the protocol defaults (as opposed to the server defaults).
+        internal IList<Http2PeerSetting> GetNonProtocolDefaults()
         {
-            yield return new Http2PeerSetting(Http2SettingsParameter.SETTINGS_HEADER_TABLE_SIZE, HeaderTableSize);
-            yield return new Http2PeerSetting(Http2SettingsParameter.SETTINGS_ENABLE_PUSH, EnablePush ? 1u : 0);
-            yield return new Http2PeerSetting(Http2SettingsParameter.SETTINGS_MAX_CONCURRENT_STREAMS, MaxConcurrentStreams);
-            yield return new Http2PeerSetting(Http2SettingsParameter.SETTINGS_INITIAL_WINDOW_SIZE, InitialWindowSize);
-            yield return new Http2PeerSetting(Http2SettingsParameter.SETTINGS_MAX_FRAME_SIZE, MaxFrameSize);
-            yield return new Http2PeerSetting(Http2SettingsParameter.SETTINGS_MAX_HEADER_LIST_SIZE, MaxHeaderListSize);
-        }
+            var list = new List<Http2PeerSetting>(0);
```

Comment:
Given that `MaxConcurrentStreams != DefaultMaxConcurrentStreams` by default (ironic no?), shouldn't we at least pass 1 into the List ctor? Also, what's wrong with IEnumerable and yield return?